### DISCUSSION TITLE
Copilot Chat - Solution Context

### DISCUSSION THREAD

**Discussion by Gareth064** [community/community#55726] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Using Visual Studio (full fat)

I have opened a Solution which contains three .NET 6 projects.

Is it possible to ask Copilot Chat questions about the whole solution, rather than asking it things about code inside a single class?

If so, then what type of prompt would you suggest?

I have asked "Can you summaries the projects in this solution" but it kept responding about the last class file I had opened.

**Comment by Akash1134** [comment, marked as answer]
Hi @Gareth064 ,

Thanks for posting on the GitHub Community.

We're still very early stages of developing Copilot Chat but this could be a great future consideration. However, at this time, Copilot (and Copilot Chat feature) cannot summarize multiple projects as it is capable to reason over one component at a time (e.g. an entire class), and does not yet have the capacity to summarize entire applications.

**Reply by Gareth064** [reply]
Thanks @Akash1134 .

Is there a thread on here or something somewhere that explains the current capabilities of it?

**Reply by Akash1134** [reply]
### From the [GitHub blog post](https://github.blog/2023-03-22-github-copilot-x-the-ai-powered-developer-experience/#a-new-ai-powered-developer-experience)

Copilot Chat will offer a ChatGPT-like experience in your editor. We are bringing a chat interface to the editor that’s focused on developer scenarios and natively integrates with VS Code and Visual Studio. This does far more than suggest code. GitHub Copilot Chat is not just a chat window. It recognizes what code a developer has typed, what error messages are shown, and it’s deeply embedded into the IDE. A developer can get in-depth analysis and explanations of what code blocks are intended to do, generate unit tests, and even get proposed fixes to bugs.

GitHub Copilot Chat builds upon the work that OpenAI and Microsoft have done with ChatGPT and the new Bing. It will also join our voice-to-code AI [technology extension we previously demoed](https://githubnext.com/projects/copilot-voice/), which we’re now calling GitHub Copilot Voice, where developers can verbally give natural language prompts.

**Reply by boberace** [reply]
[@Akash1134](https://github.com/Akash1134) Is it accurate to say that “Copilot (and Copilot Chat feature) cannot summarize multiple projects as it is capable to reason over one component at a time (e.g. an entire class)” ?  This seems highly speculative in context of an LLM.  This meaning that there would have to be a function in Copilot Chat to limit this before sending to the LLM.

Would it be more accurate to say there is simply a limited context length?  I have developed an app using GPT4 spanning multiple classes and it works fantastic when I finally understood the concept of limited token context.  I realized I had to maintain the context for each prompt as there is no way to know what context is overwritten by the chat when the limit is reached.  I honestly appears to be a circular buffer but I don’t know hence I maintain context in each prompt.

What is the token context length of Copilot Chat?  If I select more than about 4k tokens it does not upload or throws an error.

**Reply by StoyanStAtanasov** [reply]
@Akash1134 Please add a discussion in [Github blog post](https://github.blog/2023-03-22-github-copilot-x-the-ai-powered-developer-experience/#a-new-ai-powered-developer-experience) about the context, where does it come from, how to control it.

**Reply by As2tech** [reply]
Whatever, we are asking or typing from anywhere it seems they want us to be in certain guidelines. the question is, why is Github called as GitHub Copilot Voice? There must be something interesting about to ask?
